File: src/main/java/um/haberes/report/model/haberes/core/ControlDto.java
package um.haberes.report.model.haberes.core;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ControlDto {

    private Long controlId;

    private Integer anho;

    private Integer mes;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaDesde;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaHasta;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaPago;

    private String aporteJubilatorio;

    private String depositoBanco;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaDeposito;

    private BigDecimal doctorado = BigDecimal.ZERO;

    private BigDecimal maestria = BigDecimal.ZERO;

    private BigDecimal especializacion = BigDecimal.ZERO;

    private BigDecimal familiaNumerosa = BigDecimal.ZERO;

    private BigDecimal escuelaPrimaria = BigDecimal.ZERO;

    private BigDecimal escuelaSecundaria = BigDecimal.ZERO;

    private BigDecimal escuelaPrimariaNumerosa = BigDecimal.ZERO;

    private BigDecimal escuelaSecundariaNumerosa = BigDecimal.ZERO;

    private BigDecimal prenatal = BigDecimal.ZERO;

    private BigDecimal libre = BigDecimal.ZERO;

    private BigDecimal ayudaEscolar = BigDecimal.ZERO;

    private BigDecimal matrimonio = BigDecimal.ZERO;

    private BigDecimal nacimiento = BigDecimal.ZERO;

    private BigDecimal funcionDireccion = BigDecimal.ZERO;

    private BigDecimal mayorResponsabilidadPatrimonial = BigDecimal.ZERO;

    private BigDecimal polimedb = BigDecimal.ZERO;

    private BigDecimal polimedo = BigDecimal.ZERO;

    private BigDecimal montoeci = BigDecimal.ZERO;

    private BigDecimal valampo = BigDecimal.ZERO;

    private BigDecimal jubilaem = BigDecimal.ZERO;

    private BigDecimal inssjpem = BigDecimal.ZERO;

    private BigDecimal osociaem = BigDecimal.ZERO;

    private BigDecimal jubilpat = BigDecimal.ZERO;

    private BigDecimal inssjpat = BigDecimal.ZERO;

    private BigDecimal osocipat = BigDecimal.ZERO;

    private BigDecimal ansalpat = BigDecimal.ZERO;

    private BigDecimal salfapat = BigDecimal.ZERO;

    private BigDecimal minimoAporte = BigDecimal.ZERO;

    private BigDecimal maximoAporte = BigDecimal.ZERO;

    private BigDecimal mincontr = BigDecimal.ZERO;

    private BigDecimal maximo1sijp = BigDecimal.ZERO;

    private BigDecimal maximo2sijp = BigDecimal.ZERO;

    private BigDecimal maximo3sijp = BigDecimal.ZERO;

    private BigDecimal maximo4sijp = BigDecimal.ZERO;

    private BigDecimal maximo5sijp = BigDecimal.ZERO;

    private BigDecimal estadoDocenteTitular = BigDecimal.ZERO;

    private BigDecimal estadoDocenteAdjunto = BigDecimal.ZERO;

    private BigDecimal estadoDocenteAuxiliar = BigDecimal.ZERO;

    private BigDecimal horaReferenciaEtec = BigDecimal.ZERO;

    private Integer modoLiquidacionId;

    private ModoLiquidacionDto modoLiquidacion;

}

File: src/main/java/um/haberes/report/kotlin/dto/haberes/core/ControlDto.kt
package um.haberes.report.kotlin.dto.haberes.core

import com.fasterxml.jackson.annotation.JsonFormat
import java.math.BigDecimal
import java.time.OffsetDateTime

data class ControlDto(

    var controlId: Long? = null,
    var anho: Int? = null,
    var mes: Int? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaDesde: OffsetDateTime? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaHasta: OffsetDateTime? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaPago: OffsetDateTime? = null,
    var aporteJubilatorio: String? = null,
    var depositoBanco: String? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaDeposito: OffsetDateTime? = null,
    var doctorado: BigDecimal = BigDecimal.ZERO,
    var maestria: BigDecimal = BigDecimal.ZERO,
    var especializacion: BigDecimal = BigDecimal.ZERO,
    var familiaNumerosa: BigDecimal = BigDecimal.ZERO,
    var escuelaPrimaria: BigDecimal = BigDecimal.ZERO,
    var escuelaSecundaria: BigDecimal = BigDecimal.ZERO,
    var escuelaPrimariaNumerosa: BigDecimal = BigDecimal.ZERO,
    var escuelaSecundariaNumerosa: BigDecimal = BigDecimal.ZERO,
    var prenatal: BigDecimal = BigDecimal.ZERO,
    var libre: BigDecimal = BigDecimal.ZERO,
    var ayudaEscolar: BigDecimal = BigDecimal.ZERO,
    var matrimonio: BigDecimal = BigDecimal.ZERO,
    var nacimiento: BigDecimal = BigDecimal.ZERO,
    var funcionDireccion: BigDecimal = BigDecimal.ZERO,
    var mayorResponsabilidadPatrimonial: BigDecimal = BigDecimal.ZERO,
    var polimedb: BigDecimal = BigDecimal.ZERO,
    var polimedo: BigDecimal = BigDecimal.ZERO,
    var montoeci: BigDecimal = BigDecimal.ZERO,
    var valampo: BigDecimal = BigDecimal.ZERO,
    var jubilaem: BigDecimal = BigDecimal.ZERO,
    var inssjpem: BigDecimal = BigDecimal.ZERO,
    var osociaem: BigDecimal = BigDecimal.ZERO,
    var jubilpat: BigDecimal = BigDecimal.ZERO,
    var inssjpat: BigDecimal = BigDecimal.ZERO,
    var osocipat: BigDecimal = BigDecimal.ZERO,
    var ansalpat: BigDecimal = BigDecimal.ZERO,
    var salfapat: BigDecimal = BigDecimal.ZERO,
    var minimoAporte: BigDecimal = BigDecimal.ZERO,
    var maximoAporte: BigDecimal = BigDecimal.ZERO,
    var mincontr: BigDecimal = BigDecimal.ZERO,
    var maximo1sijp: BigDecimal = BigDecimal.ZERO,
    var maximo2sijp: BigDecimal = BigDecimal.ZERO,
    var maximo3sijp: BigDecimal = BigDecimal.ZERO,
    var maximo4sijp: BigDecimal = BigDecimal.ZERO,
    var maximo5sijp: BigDecimal = BigDecimal.ZERO,
    var estadoDocenteTitular: BigDecimal = BigDecimal.ZERO,
    var estadoDocenteAdjunto: BigDecimal = BigDecimal.ZERO,
    var estadoDocenteAuxiliar: BigDecimal = BigDecimal.ZERO,
    var horaReferenciaEtec: BigDecimal = BigDecimal.ZERO,
    var modoLiquidacionId: Int? = null,
    var modoLiquidacion: ModoLiquidacionDto? = null

)
